src/Action/SetObjectFieldValueAction.php

```diff
@@ -164,4 +161,32 @@ public function __invoke(Request $request): JsonResponse
 
         return new JsonResponse($content, Response::HTTP_OK);
     }
+
+    private function getFieldDataTransformer(
+        FieldDescriptionInterface $fieldDescription,
+        ModelManagerInterface $modelManager
+    ): ?DataTransformerInterface {
+        $data_transformer = $fieldDescription->getOption('data_transformer');
+
+        // Handle date type has setter expect a DateTime object
+        if ('date' === $fieldDescription->getType()) {
+            $data_transformer = new DateTimeToStringTransformer();
+        }
+
+        // Handle boolean type transforming the value into a boolean
+        if ('boolean' === $fieldDescription->getType()) {
+            $data_transformer = new BooleanToStringTransformer('1');
```

Comment:
I'm not sure that we need a date transformer here. Using a [Symfony BooleanToStringTransformer](https://github.com/symfony/symfony/blob/b3467e641c4ccaddb9bbf7c7c56b918a2f3eb635/src/Symfony/Component/Form/Extension/Core/DataTransformer/BooleanToStringTransformer.php) here may interrupt BC. At least the tests use a [non-string](https://github.com/sonata-project/SonataAdminBundle/pull/5937/files#diff-a0b7a5423dfc2dba3b9e37fd138f3d58R97).

We can use `CallbackTransformer` here, so as not to create a extra transformer.

```php
$data_transformer = new CallbackTransformer(static function ($value) {
    return (string) (int) $value;
}, static function ($value) {
    return filter_var($value, FILTER_VALIDATE_BOOLEAN);
});
```